docs/src/pages/docs/crocks/Async.md

```diff
@@ -0,0 +1,1357 @@
+will be executed concurrently.
+
+```javascript
+const Async = require('crocks/Async')
+const { Rejected, Resolved } = Async
+
+const liftA2 = require('crocks/helpers/liftA2')
+
+// log :: String -> a -> a
+const log = label => x =>
+  (console.log(`${label}:`, x), x)
+
+// resolveAfter :: (Integer, a) -> Async e a
+const resolveAfter = (delay, x) => Async(
+  (rej, res) => setTimeout(() => res(x), delay)
+)
+
+// join :: String -> String -> String
+const join =
+  a => b => `${a} ${b}`
+
+Async.of(join)
+  .ap(Resolved('blip'))
+  .ap(Resolved('blop'))
+  .fork(log('rej'), log('res'))
+//=> res: "blip blop"
+
+Async.of(join)
+  .ap(Resolved('blip'))
+  .ap(Rejected('Not Good'))
+  .fork(log('rej'), log('res'))
+//=> rej: "Not Good"
+
+Resolved('splish')
+  .map(join)
+  .ap(Resolved('splash'))
+  .fork(log('rej'), log('res'))
+//=> res: "splish splash"
+
+// first :: Async e String
+const first =
+  resolveAfter(5000, 'first')
+
+// second :: Async e String
+const second =
+  resolveAfter(5000, 'second')
+
+// `ap` runs all Asyncs at the same time in parallel.
+// This will finish running in about 5 seconds and
+// not 10 seconds
+
+liftA2(join, first, second)
+  .fork(log('rej'), log('res'))
+//=> res: "first second"
+```
+
+#### chain
+
+```haskell
+Async e a ~> (a -> Async e b) -> Async e b
+```
+
+Combining a sequential series of transformations that capture disjunction can be
+accomplished with `chain`. `chain` expects a unary, `Async` returning function
+as its argument. When invoked on a [`Rejected`](#rejected) instance , `chain` will not run
+the function, but will instead return another [`Rejected`](#rejected) instance wrapping the
+original [`Rejected`](#rejected) value. When called on a [`Resolved`](#resolved) instance however, the
+inner value will be passed to provided function, returning the result as the
+new instance.
+
+```javascript
+const Async = require('crocks/Async')
+const { Rejected, Resolved } = Async
+
+const compose = require('crocks/helpers/compose')
+const composeK = require('crocks/helpers/composeK')
+const constant = require('crocks/combinators/constant')
+const flip = require('crocks/combinators/flip')
+const ifElse = require('crocks/logic/ifElse')
+const isString = require('crocks/predicates/isString')
+const prop = require('crocks/Maybe/prop')
+const maybeToAsync = require('crocks/Async/maybeToAsync')
+
+// log :: String -> a -> a
+const log = label => x =>
+  (console.log(`${label}:`, x), x)
+
+// safe :: (b -> Boolean) -> b -> Async String a
+const safe = pred =>
+  ifElse(pred, Resolved, constant(Rejected('Not Safe')))
+
+// test :: RegExp -> String -> Boolean
+const test = regex => str =>
+  regex.test(str)
+
+// lookup :: String -> Async String String
+const lookup = compose(
+  maybeToAsync('Not Found'),
+  flip(prop, { 'file-a': 'file-b', 'file-b': 'file-c' })
+)
+
+// fake :: String -> Async String Object
+const fake = composeK(
```

Comment:
This example assumes that the reader is already familiar with what `composeK` does. Do you think being more explicit would help?

Reply:
I put function like that in these examples as teasers, do people will be like what is this `composeK` and then go look it up. Like reading a book with a Dictionary, do you think it would be more helpful to the reader to do this instead:
```js
const fake = compose(
  chain(lookup),
  chain(safe(test(/^file-(a|b|c)/)),
  chain(safe(isString))
)
```

Reply:
Sure, we can retain the `composeK`. The only reason I felt that way because the section header was `chain` but there was no reference to it in the code snippet.

Reply:
That is an excellent point! Good call! I will make that change for this one, and leave the other one as `composeK`, which will give a hint as to what the heck `composeK` could mean. What do you think?

Reply:
Sure, I think that makes sense.